src/client/executor.rs

```diff
@@ -70,6 +72,26 @@ impl Client {
         op: T,
         session: impl Into<Option<&mut ClientSession>>,
     ) -> Result<T::O> {
+        self.execute_operation_pinned(op, session, None).await
+    }
+
+    pub(crate) async fn execute_operation_pinned<T: Operation>(
```

Comment:
I'm not thrilled with `_pinned` (use passed-in connection) vs `_pinnable` (return checked-out connection), but this at least avoids the churn of updating all unrelated `execute_operation` calls.

Reply:
That cleans things up but runs into lifetime issues - with that signature it mutably borrows the `Operation` for the span that the returned connection is used, which conflicts with the rest of the execution.  `Operation` can be parameterized to give the return a `&'conn mut Connection` lifetime to disconnect the borrows, but that has substantial ripple effects on the rest of the driver code.  Sadly, default lifetimes never made it in, because `Operation<'conn = 'static>` would be ideal here.